fake_switches/juniper/juniper_netconf_datastore.py

```diff
@@ -210,10 +217,11 @@ def parse_interfaces(self, conf, etree_conf):
             operation = resolve_operation(interface_node)
             if operation == "delete" and isinstance(port, AggregatedPort):
                 conf.remove_port(port)
+            if operation == "delete" and not isinstance(port, AggregatedPort) and not isinstance(port, VlanPort):
```

Comment:
Not sure about this: 
1- The VlanPort case is not handled.
2- I dont like the way we are listing subclasses to know if this is the parent class (easily forgotten in case of adding a class)
Improvement suggestion, VlanPort is handled the same way as a Port 

    elif operation=="delete":
        port.reset()